### PR TITLE
[SC-406] Optional Max DSR

### PR DESCRIPTION
**Pull request by hexonaut**
On L2s it makes sense to completely trust the bridge and just burn admin access. As such we have been setting the max dsr to an arbitrarily high value (1000% / year). While this is fine in practise, it is an improvement to allow for the oracle to just disable this feature explicitly. This allows for longer durations for `rpow` computation before it overflows. With 1000% APY you have an overflow if no update is performed within 20 years which again is fine in practise, but we can remove this limiter.

### CODE REVIEW

File: test/DSRAuthOracle.t.sol [hunk 1]
```diff
@@ -33,6 +32,10 @@ contract DSRAuthOracleTest is Test {
         skip(1 * (365 days));
     }
 
+    function test_constructor() public {
+        assertEq(oracle.maxDSR(), 0);
+    }
+
```
Comment (lucas-manuel) [inline test/DSRAuthOracle.t.sol:38]: Need full testing for `setMaxDSR`:
- Permissioning
- `< RAY` boundary condition
- `>= RAY` condition
- 0 condition

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by barrutko — COMMENTED** [review]
Agreed with Lucas' comment.
Other than that LGTM.
Will approve after the changes.

File: test/DSRAuthOracle.t.sol (hunk 1 above)
Comment (hexonaut) [reply test/DSRAuthOracle.t.sol:38]: added

**Review by barrutko — DISMISSED** [review]

**Review by lucas-manuel — DISMISSED** [review]

**Review by barrutko — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]
